### ISSUE TITLE
Public viewer

### ISSUE THREAD

**Issue by christianbundy** [fraction/oasis#48]
**What's the problem you want to solved?** It's fun to run a pub + viewer but hard to do with Oasis. I have a branch called `public`, but it's not up-to-date with `master`.

**Is there a solution you'd like to recommend?** Maybe we should have support for Oasis as a public viewer? Would anyone want to use this?

**Comment by christianbundy** [comment]
Now that the demo is outdated, this is more important than ever. :~)

**Comment by cinnamon-bun** [comment]
Features needed for this (please let me know if I misinterpreted)

- [ ] Add command line flag for "public viewer" mode, which:
  - [ ] Makes everything read-only
  - [ ] Hides feeds which haven't opted in to public viewing
  - [ ] Sets more cache-y cache headers (in case the server gets a lot of traffic)
  - [ ] ❓robots.txt to exclude search engines?
- [ ] Add a button to your profile to opt in or out of public viewing by publishing a message to your feed; also show your opt-in status there.  (Default is opt-out.)

What if there was also `blog mode` - a public viewer that's focused around one person's feed, so you could put your own content online like a blog.  It would be limited to just your profile, maybe even just your own root-level posts with no comments to minimize privacy issues.

I'm wary of public viewers; people tend to quote each other in messages without realizing they might be publishing the content they quote to a public viewer if they themselves have opted in.
Mitigations:
- [ ] One-person public viewer which doesn't show comments, as above
- [ ] If you've opted in to a public viewer, make this obvious when you're writing a message or looking at your own profile
- [ ] Show if other people have opted in on their profile pages (it can be less obvious there)

**Comment by christianbundy** [comment]
Thanks for speccing this out! That's really useful, and it's especially the kind of work that I have a hard time with sometimes. :heart: 

> I'm wary of public viewers; people tend to quote each other in messages without realizing they might be publishing the content they quote to a public viewer if they themselves have opted in.

I'm very conflicted about `publicWebHosting` in general. I've talked about this a few times on SSB, but I have lots of anxiety around someone mistaking the `publicWebHosting` option for any sort of real privacy, and I'm scared that we're normalizing this "you're safe here" feeling when posting something permanently on the internet.

I have this gut feeling that public viewers help bridge that gap, and make our lack of privacy harder to ignore, but I'm also open to the idea that my gut feeling is wrong and that public viewers like oasis-demo are actively harmful or dangerous.

I think I'm also having a hard time understanding exactly what "public web hosting" means, but I know that sometimes I struggle with fuzzy boundaries and I try to make them sharp lines. I tried to reconcile that a bit [here](https://oasis-demo.fraction.io/thread/%255SpWGMchnWuOcjKI87C0lokfzRZv%2FTu1EB9n%2BDhH%2B4A%3D.sha256#%255SpWGMchnWuOcjKI87C0lokfzRZv%2FTu1EB9n%2BDhH%2B4A%3D.sha256) but didn't get to a place where I understood. For example, I'm not sure whether [encoding SSB feeds as Git branches](https://github.com/christianbundy/ssb-git-data) and putting them on GitHub would be a violation of `publicWebHosting`, and thinking about quoting people makes the problem feel murkier and harder to reason about.

> - One-person public viewer which doesn't show comments, as above

I'd really love this, and I'd like to build it in the future. I think this is adjacent to a public viewer, but it's still a great thing (and I've been wanting this for a long time). 

> - If you've opted in to a public viewer, make this obvious when you're writing a message or looking at your own profile
> - Show if other people have opted in on their profile pages (it can be less obvious there)

Super into these as well.
